libraries/AP_Common/Location.h

```diff
@@ -14,10 +14,21 @@
 class AP_AHRS_NavEKF;
 class AP_Terrain;
 
-class Location_Class : public Location
+class Location
 {
 public:
 
+    uint8_t relative_alt : 1;           // 1 if altitude is relative to home
```

Comment:
Have you tried not constraining these? I'm guessing you get a non trivial size savings, and faster access. (Increases memory consumption admittedly will go up)

Reply:
Definitely want to investigate doing that as a separate PR.  I've done that before and the code savings are significant.